compiler/modules/nback/build.bal

```diff
@@ -948,8 +948,210 @@ function buildCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn i
+                    if insn.op is "<=" || insn.op is ">=" {
+                        buildStoreBoolean(builder, scaffold, bothNull, insn.result);
+                    }
+                    else {
+                        buildStoreBoolean(builder, scaffold, llvm:constInt(LLVM_BOOLEAN,0), insn.result);
+                    }
+                    builder.br(joinBB);
+
+                    builder.positionAtEnd(neitherNullBB);
+                    if lhs is () || rhs is () {
+                        builder.unreachable();
+                    }
+                    else {
+                        check untagAndCompare(builder, scaffold, insn);
+                        builder.br(joinBB);
+                    }
+                    
+                    builder.positionAtEnd(joinBB);
+                }
+                else {
+                    // Only one operand can be null 
+                    bir:Operand nullableOperand;
+                    llvm:Value nullableValue;
+                    bir:Operand other;
+                    llvm:Value otherValue;
+                    if lhs is () || (lhs is bir:Register && t:containsNil(lhs.semType)) {
+                        nullableOperand = lhs;
+                        nullableValue = lhsVal;
+                        other = rhs;
+                        otherValue = rhsVal;
+                    }
+                    else {
+                        nullableOperand = rhs;
+                        nullableValue = rhsVal;
+                        other = lhs;
+                        otherValue = lhsVal;
+                    }
+                    llvm:Value isNull = builder.iCmp("eq", nullableValue, llvm:constNull(llvm:pointerType("i8", 1)));
+                    llvm:BasicBlock isNullBB = scaffold.addBasicBlock();
+                    llvm:BasicBlock notNullBB = scaffold.addBasicBlock();
+                    llvm:BasicBlock joinBB = scaffold.addBasicBlock();
+                    builder.condBr(isNull, isNullBB, notNullBB);
+
+                    builder.positionAtEnd(isNullBB);
+                    buildStoreBoolean(builder, scaffold, llvm:constInt(LLVM_BOOLEAN,0), insn.result);
+                    builder.br(joinBB);
+
+                    builder.positionAtEnd(notNullBB);
+                    if lhs is () || rhs is () {
+                        builder.unreachable();
+                    }
+                    else {
+                        check untagAndCompare(builder, scaffold, insn);
+                        builder.br(joinBB);
+                    }
+                    
+                    builder.positionAtEnd(joinBB);
+                }
+            }
+            else {
+                return err:unimplemented("compare with operands of non-optional unimplemented type");
+            }
+        }
+    }
+}
+
+function loadOperandForOptionalComparison(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn insn, bir:Operand operand) returns llvm:Value|BuildError {
```

Comment:
Choose a function name consistent with other names in this file.

Reply:
Function removed